modules/basicfuncs/numeric-funcs.c

```diff
@@ -130,3 +132,156 @@ tf_num_mod(LogMessage *msg, gint argc, GString *argv[], GString *result)
 }
 
 TEMPLATE_FUNCTION_SIMPLE(tf_num_mod);
+
+
+static GString *
+_make_gstring_buffer(const LogTemplateInvokeArgs *args)
+{
+  if (args->bufs->len == 0)
+    g_ptr_array_add(args->bufs, g_string_sized_new(64));
+
+  return (GString *) g_ptr_array_index(args->bufs, 0);
+}
+
```

Comment:
This function should probably go to lib/template somehow. It is clearly operating on LogTemplateInvokeArgs, eg it seems to smell from feature envy